src/commands/moderation/repel.ts

```diff
@@ -0,0 +1,402 @@
+  return target.communicationDisabledUntilTimestamp
+    ? target.communicationDisabledUntilTimestamp > Date.now()
+    : false;
+};
+
+const checkCanRepel = ({
+  commandUser,
+  repelRole,
+}: {
+  commandUser: GuildMember;
+  repelRole: Role;
+}): {
+  ok: boolean;
+  message?: string;
+} => {
+  const hasPermission =
+    commandUser.permissions.has(PermissionFlagsBits.ModerateMembers) ||
+    commandUser.roles.cache.has(repelRole.id);
+  if (!hasPermission) {
+    return { ok: false, message: 'You do not have permission to use this command.' };
+  }
+  return { ok: true };
+};
+
+const checkCanRepelTarget = ({
+  target,
+  commandUser,
+  botMember,
+}: {
+  target: User | GuildMember;
+  commandUser: GuildMember;
+  botMember: GuildMember;
+}): {
+  ok: boolean;
+  message?: string;
+} => {
+  if (!isUserInServer(target)) {
+    return { ok: true };
+  }
+
+  if (target.user.bot) {
+    return { ok: false, message: 'You cannot repel a bot.' };
+  }
+
+  if (target.id === commandUser.id) {
+    return { ok: false, message: 'You cannot repel yourself.' };
+  }
+
+  const isTargetServerOwner = target.id === target.guild.ownerId;
+  if (isTargetServerOwner) {
+    return { ok: false, message: 'You cannot repel the server owner.' };
+  }
+
+  if (target.roles.highest.position >= commandUser.roles.highest.position) {
+    return { ok: false, message: 'You cannot repel this user due to role hierarchy.' };
+  }
+
+  if (target.roles.highest.position >= botMember.roles.highest.position) {
+    return { ok: false, message: 'I cannot repel this user due to role hierarchy.' };
+  }
+
+  return { ok: true };
+};
+
+const getTargetFromInteraction = async (
+  interaction: ChatInputCommandInteraction
+): Promise<GuildMember | User> => {
+  const targetFromOption = interaction.options.getUser(RepelOptions.TARGET, true);
+  let target: User | GuildMember | null = null;
+  if (!interaction.inGuild() || interaction.guild === null) {
+    return targetFromOption;
+  }
+  try {
+    target = await interaction.guild.members.fetch(targetFromOption.id);
+  } catch {
+    target = targetFromOption;
+  }
+  return target;
+};
+
+const handleTimeout = async ({
+  target,
+  duration,
+}: {
+  target: GuildMember | User;
+  duration: number;
+}): Promise<number> => {
+  if (duration === 0 || !isUserInServer(target) || isUserTimedOut(target)) {
+    return 0;
+  }
+  try {
+    await target.timeout(duration * HOUR, 'Repel command executed');
+    return duration;
+  } catch {
+    return 0;
+  }
+};
+
+const getTextChannels = (interaction: ChatInputCommandInteraction) => {
+  if (!interaction.inGuild() || !interaction.guild) {
+    return [];
+  }
+  const channels = getPublicChannels(interaction.guild).map((c) => c);
+  return channels;
+};
+
+const handleDeleteMessages = async ({
+  target,
+  channels,
+  lookBack,
+}: {
+  target: GuildMember | User;
+  channels: TextChannel[];
+  lookBack: number;
+}) => {
+  let deleted = 0;
+  await Promise.all(
```

Comment:
use `Promise.allSettled` over `Promise.all`. In Promise.all if one call gets rejected, everything fails.

`allSettled` will return an array of the results, both successful and rejected. count the failures and log it. maybe also with a list of which channels failed if that's somehow possible. i don't know what the error message could look like, so i can't be more specific.

```ts
console.error(`Failed to delete messages in ${amountFailedChannels} channels.`)
```